### ISSUE TITLE
KeyError: 'Localising' for some semiology terms

### ISSUE THREAD

**Issue by fepegar** [thenineteen/Semiology-Visualisation-Tool#7]
Such as `Whistling`.

**Comment by thenineteen** [comment]
you'll need to specify which file throws this error please

**Comment by fepegar** [comment]
I can't debug because of [this issue](https://github.com/thenineteen/Semiology-Visualisation-Tool/issues/18), so that needs to be solved first.

**Comment by fepegar** [comment]
```python-traceback
$ python -c "import mega_analysis; mega_analysis.get_scores_dict('test', catch_errors=False)"
Scores dictionary for semiology term test not retrieved:
'Localising'
Traceback (most recent call last):
  File "/Users/fernando/miniconda3/envs/ali/lib/python3.6/site-packages/pandas/core/indexes/base.py", line 2897, in get_loc
    return self._engine.get_loc(key)
  File "pandas/_libs/index.pyx", line 107, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 131, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1607, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1614, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'Localising'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/fernando/git/Semiology-Visualisation-Tool/mega_analysis/scores.py", line 122, in get_scores_dict
    output_path=output_path,
  File "/Users/fernando/git/Semiology-Visualisation-Tool/mega_analysis/scores.py", line 68, in get_scores
    semiology_dict_path=semiology_dict_path,
  File "/Users/fernando/git/Semiology-Visualisation-Tool/mega_analysis/crosstab/mega_analysis/QUERY_SEMIOLOGY.py", line 234, in QUERY_SEMIOLOGY
    logging.debug(f'\nLocalising Datapoints relevant to query {semiology_term}: {inspect_result["Localising"].sum()}')
  File "/Users/fernando/miniconda3/envs/ali/lib/python3.6/site-packages/pandas/core/frame.py", line 2995, in __getitem__
    indexer = self.columns.get_loc(key)
  File "/Users/fernando/miniconda3/envs/ali/lib/python3.6/site-packages/pandas/core/indexes/base.py", line 2899, in get_loc
    return self._engine.get_loc(self._maybe_cast_indexer(key))
  File "pandas/_libs/index.pyx", line 107, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 131, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1607, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1614, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'Localising'
```

This message is not informative.
A `KeyError` should be manually raised if the semiology term is not found, with a better message such as "`"test" is not a valid semiology term`".

**Comment by thenineteen** [comment]
works for me when running the module separately from QUERY_SEMIOLOGY
as can be seen, there clearly is a 'Localising' column in the DataFrame so not sure yet where the problem is here:

![image](https://user-images.githubusercontent.com/10786099/77834855-cc23d480-713f-11ea-80b3-b9c71447af0c.png)

**Comment by fepegar** [comment]
I don't get the error with `Whistling` anymore. But in my example above, I used a made-up semiology (`test`). I do get the error with that one.

**Comment by thenineteen** [comment]
if you use a made up semiology, you need to ensure you've set semiology_dict_path to None.

**Comment by fepegar** [comment]
Well, the point is that I don't know if the semiology is in the dataset or not, so I can't decide the value for `semiology_dict_path`. The software must be robust to whether the semiology term exists or not.

**Comment by thenineteen** [comment]
fair point

**Comment by thenineteen** [comment]
please confirm the fix works for you? neat - I didn't realise a commit message could automatically close an issue!

**Comment by fepegar** [comment]
It doesn't. Please don't close the issue if there's no confirmation that it has been solved. Also, can you start creating pull requests before committing directly to master? That way we can discuss the changes and tests can be run automatically once you've enabled the CI on Travis.

```bash
$ python -c "import mega_analysis; scores = mega_analysis.get_scores_dict('dsf', catch_errors=False); print(scores)"
```

```python-traceback
QUERY SEMIOLOGY ERROR: This semiology was not found within the reported literature nor in the semiology categories
Scores dictionary for semiology term dsf not retrieved:
'NoneType' object is not subscriptable
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/fernando/git/Semiology-Visualisation-Tool/mega_analysis/scores.py", line 130, in get_scores_dict
    output_path=output_path,
  File "/Users/fernando/git/Semiology-Visualisation-Tool/mega_analysis/scores.py", line 85, in get_scores
    pts_dominant_hemisphere_R_or_L=dominant_hemisphere,
  File "/Users/fernando/git/Semiology-Visualisation-Tool/mega_analysis/crosstab/mega_analysis/QUERY_LATERALISATION.py", line 53, in QUERY_LATERALISATION
    Lat = inspect_result['Lateralising']
TypeError: 'NoneType' object is not subscriptable
```

**Comment by fepegar** [comment]
It actually makes it worse, you introduced an infinite loop so the program just hangs. I'm going to revert that commit.

**Comment by thenineteen** [comment]
well this is technically a TypeError now and from a different file... we could deal with this:

1. either by sys.exit()
2. or fillna(0) and display an empty brain

**Comment by thenineteen** [comment]
ok done - please check pull request

**Comment by fepegar** [comment]
Reopening this issue as it was closed because "fix #7" was in the text of the reverting commit.

**Comment by fepegar** [comment]
I can give it a try if you like.
